coderd/database/modelqueries.go

```diff
@@ -0,0 +1,83 @@
+package database
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+)
+
+// customQuerier encompasses all non-generated queries.
+// It provides a flexible way to write queries for cases
+// where sqlc proves inadequate.
+type customQuerier interface {
+	templateQuerier
+}
+
+type templateQuerier interface {
+	UpdateTemplateUserACLByID(ctx context.Context, id uuid.UUID, acl UserACL) error
+	GetTemplateUserRoles(ctx context.Context, id uuid.UUID) ([]TemplateUser, error)
+}
+
+type TemplateUser struct {
+	User
+	Role TemplateRole `db:"role"`
+}
+
+func (q *sqlQuerier) UpdateTemplateUserACLByID(ctx context.Context, id uuid.UUID, acl UserACL) error {
+	raw, err := json.Marshal(acl)
+	if err != nil {
+		return xerrors.Errorf("marshal user acl: %w", err)
+	}
+
+	const query = `
+UPDATE
+	templates
+SET
+	user_acl = $2
+WHERE
+	id = $1`
+
+	_, err = q.db.ExecContext(ctx, query, id.String(), raw)
+	if err != nil {
+		return xerrors.Errorf("update user acl: %w", err)
+	}
+
+	return nil
+}
+
+func (q *sqlQuerier) GetTemplateUserRoles(ctx context.Context, id uuid.UUID) ([]TemplateUser, error) {
+	const query = `
```

Comment:
I'm not convinced we should escape sqlc here... this function is only used once, so why not just do the struct conversion where it's queried from instead?

Reply:
there's unfortunately multiple problems when I tried to write this with `sqlc`. One is that it didn't recognize the intermediate `value` column in the subquery. When I tried to jank around that the resulting return type was wildly different than what I tried to express :(